org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/jdt/JavaDebugServer.java

```diff
@@ -23,90 +23,118 @@
 import org.eclipse.jdt.ls.debug.internal.Logger;
 
 public class JavaDebugServer implements IDebugServer {
```

Comment:
JavaDebugServer extends the eclipse extension point, and eclipse will new a instance using the default constructor every time, so use the internal class to implement singleton pattern. However, i found another usage about extension point by using Extension Factory, see my new change set. It implements singleton pattern in JavaDebugServer directly, and let eclipse invoke a factory (IExecutableExtensionFactory#create()) to create extension instance.